cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,639 @@
+import itertools
+from typing import Sequence, Tuple, Iterator
+from cirq import circuits, devices, ops, protocols, sim, study, value
+import numpy as np
+from matplotlib import pyplot
+from mpl_toolkits.mplot3d import Axes3D
+
+
+class RabiResult:
+    """Results from a Rabi oscillation experiment."""
+
+    def __init__(self, rabi_angles: Sequence[float],
+                 excited_state_probabilities: Sequence[float]):
+        """
+        Args:
+            rabi_angles: The rotation angles of the qubit around the x-axis
+                of the Bloch sphere.
+            excited_state_probabilities: The corresponding probabilities that
+                the qubit is in the excited state.
+        """
+        self._rabi_angles = rabi_angles
+        self._excited_state_probs = excited_state_probabilities
+
+    @property
+    def data(self) -> Sequence[Tuple[float, float]]:
+        """Returns a sequence of tuple pairs with the first item being a Rabi
+        angle and the second item being the corresponding excited state
+        probability.
+        """
+        return [(angle, prob) for angle, prob in zip(self._rabi_angles,
+                                                     self._excited_state_probs)]
+
+    def plot(self) -> None:
+        """Plots excited state probability vs the Rabi angle (angle of rotation
+        around the x-axis).
+        """
+        fig = pyplot.figure()
+        pyplot.plot(self._rabi_angles, self._excited_state_probs, 'ro-',
```

Comment:
When I used this method, my first reaction was "I want the points to be smaller". I think we can very hackily give control over that kind of thing by taking a **kwargs argument, and forwarding it into the `plot` call. Then users could say `results.plot(color='black')` or `results.plot(markersize=0.5, linewidth=0.2)` or etc.

Reply:
Yeah this sounds like a good idea. Done.